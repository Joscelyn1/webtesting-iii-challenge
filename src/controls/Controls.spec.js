import React from 'react';
import renderer from 'react-test-renderer'; // 1: install this npm module as a dev dependency
import { render, fireEvent, getByTestId } from '@testing-library/react';
import Dashboard from '../dashboard/Dashboard.js';
import Controls from './Controls';

describe('<Controls />', () => {
  // 2. write this test
  it('matches snapshot', () => {
    const tree = renderer.create(<Controls />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("defaults to 'unlocked' and 'open'", () => {
    const { getByText } = render(<Dashboard />);
    expect(getByText('Locked Gate'));
    expect(getByText('Closed Gate'));
    expect(getByText('Open'));
    expect(getByText('Unlocked'));
  });

  it('cannot be closed or opened if it is locked', () => {
    const { getByText, queryByText } = render(<Dashboard />);

    const openCloseButton = getByTestId('openClose');

    fireEvent.click(openCloseButton);
  });
});

/*
- defaults to `unlocked` and `open`
- cannot be closed or opened if it is locked
*/
//lockUnlock
//openClose
