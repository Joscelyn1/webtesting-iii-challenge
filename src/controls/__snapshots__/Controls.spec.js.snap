// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lockUnlock"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="openClose"
  >
    Close Gate
  </button>
</div>
`;

exports[`<Speak /> matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lockUnlock"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="openClose"
  >
    Close Gate
  </button>
</div>
`;
